File: packages/point/python_std/point_std/fs.py
from __future__ import annotations

from pathlib import Path


def readFile(path: str) -> str | dict[str, str]:
	try:
		return Path(path).read_text(encoding="utf-8")
	except Exception as error:
		return {"message": str(error)}


def writeFile(path: str, contents: str) -> None | dict[str, str]:
	try:
		Path(path).write_text(contents, encoding="utf-8")
		return None
	except Exception as error:
		return {"message": str(error)}

File: packages/point/runtime/builtins/fs.ts
import { readFileSync, writeFileSync } from "node:fs";

export type PointRuntimeFsError = { message: string };

export function readFile(path: string): string | PointRuntimeFsError {
	try {
		return readFileSync(path, "utf8");
	} catch (error) {
		return { message: error instanceof Error ? error.message : String(error) };
	}
}

export function writeFile(path: string, contents: string): null | PointRuntimeFsError {
	try {
		writeFileSync(path, contents, "utf8");
		return null;
	} catch (error) {
		return { message: error instanceof Error ? error.message : String(error) };
	}
}
